src/content/posts/2023-10-24-git-workflow.md

```diff
@@ -0,0 +1,132 @@
+---
+title: "Tips to git good with git"
+slug: git-workflow
+date: 2023-10-24 12:00:00
+tags:
+- git
+categories:
+- git
+twitter_text: "Some helpful tips and commands to manage your git workflow."
+authors: 
+- Jaime Jones
+image: https://images.unsplash.com/photo-1604918598683-28d4e6e2986d?auto=format&fit=crop&q=80&w=3570&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D
+image_url: https://unsplash.com/photos/black-bare-tree-under-white-sky-during-daytime-Ku-1SYS0o7k
+image_credit: Mila Tovar
+---
+
+Whether you love it or hate it, it's undeniable that git (or your source control of choice, but this post is about git, so git on outta here if you use something else) is an integral part of our lives as developers. There are a lot of ways to manage respositories and codebases, but the [forking workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow) is the workflow that has worked for me and for our team with only minimal nuclear level resets required. Who here has had to delete and refork a repository and then clone it down fresh again? If that's been you (don't worry, it's probably been all of us at some point), maybe this post can help.
+
+### Managing repositories
+
+##### The commands
+- `git clone {url}` - clones down a respository
+- `git remote add {name} {url}` - sets a new remote with the given name
+
+##### The workflow
+- Fork the repository from your organization to your own github account
+- Clone down your fork of the repository using `git clone {fork url}`
+- Add your organization's copy of the repository as a remote with the desired name (we always use `upstream`) using `git remote add {name} {main repo url}`
+
+##### The general idea
+
+You should always have your own fork of your organization's repository that you can do your work from. This isolates your work from the organization's codebase which adds a layer of protection. If you use `upstream` as the name of the organization's copy of the repository, you'll end up with two remotes (the codebase stored on github) to manage. You'll have `origin` which is your copy and you'll have `upstream` which is the organization's copy.
+
+### Navigating branches
+
+##### The commands
+- `git checkout {branch}` - moves to an existing branch of the provided name
+- `git checkout -b {branch}` - creates a new branch with the provided name, using the current branch as the base
+- `git branch -d {branch}` - deletes the branch with the provided name from your local machine
+- `git push origin :{branch}` - deletes the branch from your github
+
+##### The workflow
+- When creating a new feature branch, you'll want to create it off of your master (or feature) branch, like so:
+```
+git checkout master
+git checkout -b my-feature
+```
```

Comment:
Here it might be good to add something like
> Be sure to make sure you have the latest version of your master branch before creating your feature branch.

Reply:
I did make some note of that in that overall section, should I still make a note about it here?